### ISSUE TITLE
*.js is excluding more files than expected

### ISSUE THREAD

**Issue by JohnnyWalkerDigital** [PHPCSStandards/PHP_CodeSniffer#1405]
## Describe the bug

Apologies if this is not a bug, but I do not understand what is happening here. I have an `exclude-pattern` rule that is excluding a file, and I cannot understand why.

This rule:

`<exclude-pattern>*.js</exclude-pattern>`

Leads to this file being excluded:

`<file>app/Services/JsonLdProcessingService.php</file>`

I have read [the documentation](https://github.com/squizlabs/PHP_CodeSniffer/wiki/Advanced-Usage#ignoring-files-and-folders) and I understand that the `exclude-pattern` uses regex, and also that `*` is converted to `.*` for convenience. I also understand that `.` should be escaped if you wish it to be treated like a literal dot (so my rule is flawed). 

However the regex pattern `/.*.js/` still does not match `app/Services/JsonLdProcessingService.php` so what is going on?

### Custom ruleset
```xml
<?xml version="1.0"?>
<ruleset name="My Custom Standard">
    <rule ref="PSR12"/>
    <file>app/Services/JsonLdProcessingService.php</file>
    <exclude-pattern>*.js</exclude-pattern>
</ruleset>
```

### To reproduce
1. Create a file called `app/Services/JsonLdProcessingService.php` with the code sample above...
2. Run `vendor/bin/phpcs -vv app/Services/JsonLdProcessingService.php`
3. See error message displayed
```text
ERROR: No files were checked.
All specified files were excluded or did not match filtering rules.
```

## Expected behavior

I'd expect the file to be checked?

## Versions (please complete the following information)

|                                          |                                                                                                                |
|-----------------------------|---------------------------------------------------------------------------- |
| Operating System            | macOS Tahoe 26.4                                                           |
| PHP version                      | 8.4.19                                                                                        |
| PHP_CodeSniffer version | 4.0.1                                                                            |
| Standard                          | PSR12                                                         |
| Install type                       | Composer (global) |

## Additional context

Apologies if this is my misunderstanding!

## Please confirm

- [*] I have searched the issue list and am not opening a duplicate issue.
- [*] I have read the [Contribution Guidelines](https://github.com/PHPCSStandards/PHP_CodeSniffer/blob/HEAD/.github/CONTRIBUTING.md) and this is not a [support question](https://github.com/PHPCSStandards/PHP_CodeSniffer/discussions).
- [*] I confirm that this bug is a bug in PHP_CodeSniffer and not in one of the external standards.
- [*] I have verified the issue still exists in the `4.x` branch of PHP_CodeSniffer.

**Comment by jrfnl** [comment]
@JohnnyWalkerDigital Thanks for doing your own research before posting. However, this is not a bug.

> However the regex pattern `/.*.js/` still does not match `app/Services/JsonLdProcessingService.php` so what is going on?

Actually it does match, as the file matching is done in a case-insensitive manner. Once you add the `i` flag, you can see that the regex matches the filename.

https://regex101.com/?regex=.*.js&testString=app%2FServices%2FJsonLdProcessingService.php&flags=gmi&flavor=pcre2&delimiter=%7E

Looking at the Wiki, I think [the docs](https://github.com/PHPCSStandards/PHP_CodeSniffer-documentation/blob/main/wiki/Advanced-Usage.md#ignoring-files-and-folders) can be improved by mentioning the case-insensitivity. Would you like to submit a PR for that ?
